client/src/components/PassReset.tsx

```diff
@@ -0,0 +1,187 @@
+      msg += err === "" ? err : `${err}.\n`;
+    }
+    msg = msg.slice(0, -1);
+    return msg;
+  };
+
+  /**
+   * Validates all input forms except for `confirmPass` input
+   * @param regexPattern  pattern to match
+   * @param checkString   string to verify
+   * @param setHook       set function for the input's hook
+   * @param errorMessage  error message to display for that input field
+   */
+  const validateBlur = (
+    regexPattern: RegExp,
+    checkString: string,
+    setHook: Function,
+    errorMessage: string
+  ) => {
+
+    if (!regexPattern.test(checkString)) {
+      setHook(errorMessage);
+    } else {
+      setHook("");
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault(); // no refresh; default
+
+    const msg = getErrorMessages();
+    if (msg !== "") {
+      console.warn(msg);
+      props.showAlert("warning", msg);
+      return;
+    }
+
+    if (isFormMissingFields()) {
+      const msg = "Please fill in all the fields.";
+      console.warn(msg);
+      props.showAlert("warning", msg);
+      return;
+    }
+
+    try {
+      const { status, data } = await api.sendPassReset(passResetForm);
+      if (status !== 204) {
+        if (!data) {
+          throw new Error("Missing error response.");
+        }
+        const msg = data.message;
+        console.error(msg);
+        props.showAlert("error", msg);
+      } else {
+        props.showAlert("success", "Password reset email sent!")
+      }
+    } catch (error) {
+      console.error(error);
+      props.showAlert(
+        "error",
+        "An error occurred logging you in. Please try again later."
+      );
+    }
+  };
+
+  const handleFormInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPassResetForm({ ...passResetForm, [e.target.name]: e.target.value });
+  };
+
+  const handleError =
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>) =>
+    (msg: string) => {
+      return setPassResetFormErrors({
+        ...passResetFormErrors,
+        [e.target.name]: msg,
+      });
+    };
+
+  return (
+    <Box
+      data-testid="ResetPassTab"
+      sx={{
+        mx: 2,
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+      }}
+      overflow="auto"
+      maxHeight="70vh"
+    >
+      <Avatar sx={{ m: 1, bgcolor: "secondary.main" }}>
+        <LockOutlinedIcon />
+      </Avatar>
+      <Typography component="h1" variant="h5">
+        Reset Password
+      </Typography>
+      <Box
+        component="form"
+        noValidate
+        autoComplete="off"
+        onSubmit={handleSubmit}
+        data-testid="form"
+        sx={{ mt: 1 }}
+      >
+        <TextField
+          name="email"
+          label="Email Address"
+          onChange={handleFormInput}
+          fullWidth
+          required
+          data-testid="resetField"
+          type="email"
+          onBlur={(e) =>
+            validateBlur(
```

Comment:
Do we really need to validate this? We could have something like "No user exists with that email", or do what a lot of sites do and say "If an account is associated to that email, then a password reset has been sent" rather than client validators

Reply:
The validation is just for correct email format, the component already implements the "No user exists with that email" message.